js/cards/disk2.ts

```diff
@@ -976,42 +951,43 @@ export default class DiskII implements Card<State>, MassStorage<NibbleFormat> {
 
     private insertDisk(drive: DriveNumber, disk: FloppyDisk) {
         const cur = this.drives[drive];
-        Object.assign(cur, disk);
+        cur.disk = disk as WozDisk | NibbleDisk;
+        cur.metadata = disk.metadata;
         const { name, side } = cur.metadata;
         this.updateDirty(drive, true);
         this.callbacks.label(drive, name, side);
     }
 
     // TODO(flan): Does not work with WOZ or D13 disks
-    getBinary(drive: DriveNumber, ext?: NibbleFormat): MassStorageData | null {
+    getBinary(drive: DriveNumber, ext?: Exclude<NibbleFormat, 'woz' | 'd13'>): MassStorageData | null {
         const cur = this.drives[drive];
-        if (!isNibbleDrive(cur)) {
+        if (!isNibbleDisk(cur.disk)) {
             return null;
         }
-        const { format, readOnly, tracks, volume } = cur;
+        const { format, readOnly, tracks, volume } = cur.disk;
         const { name } = cur.metadata;
         const len = format === 'nib' ?
             tracks.reduce((acc, track) => acc + track.length, 0) :
             this.sectors * tracks.length * 256;
         const data = new Uint8Array(len);
 
-        ext = ext ?? format;
+        const extension = ext ?? format;
         let idx = 0;
         for (let t = 0; t < tracks.length; t++) {
             if (ext === 'nib') {
                 data.set(tracks[t], idx);
                 idx += tracks[t].length;
             } else {
                 for (let s = 0; s < 0x10; s++) {
-                    const sector = readSector({ ...cur, format: ext }, t, s);
+                    const sector = readSector({ ...cur.disk, format: extension }, t, s);
                     data.set(sector, idx);
                     idx += sector.length;
                 }
             }
         }
 
         return {
-            ext,
+            ext: extension,
```

Comment:
We should probably settle on `ext` or `extension` so we don't have to not use shorthand.

Reply:
BTW, my current draft of `d13` support is uploaded at https://github.com/whscullin/apple2js/commit/2c05fe59ccf9a99b6b6237467371cce0932c07b3.

Let me know how you would like to proceed.